src/core/public/application/application_service.tsx

```diff
@@ -255,10 +255,10 @@ export class ApplicationService {
         }
         if (openInNewTab) {
           this.openInNewTab!(getAppUrl(availableMounters, appId, path));
-        } else {
-          this.navigate!(getAppUrl(availableMounters, appId, path), state, replace);
+          return;
         }
 
+        this.navigate!(getAppUrl(availableMounters, appId, path), state, replace);
```

Comment:
Yea, `currentAppId$` shouldn't emit when we're opening in a new tab. 

```ts
this.appInternalStates.delete(this.currentAppId$.value!);
```

should not be triggered either and should also be skipped. Could you please also fix this?


```patch
Index: src/core/public/application/application_service.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/core/public/application/application_service.tsx b/src/core/public/application/application_service.tsx
--- a/src/core/public/application/application_service.tsx	(revision 501e3150111867f0a100b49daf52463d05a12069)
+++ b/src/core/public/application/application_service.tsx	(date 1625645624318)
@@ -250,16 +250,15 @@
         if (path === undefined) {
           path = applications$.value.get(appId)?.defaultPath;
         }
-        if (!navigatingToSameApp) {
-          this.appInternalStates.delete(this.currentAppId$.value!);
-        }
         if (openInNewTab) {
           this.openInNewTab!(getAppUrl(availableMounters, appId, path));
         } else {
+          if (!navigatingToSameApp) {
+            this.appInternalStates.delete(this.currentAppId$.value!);
+          }
           this.navigate!(getAppUrl(availableMounters, appId, path), state, replace);
-        }
-
-        this.currentAppId$.next(appId);
+          this.currentAppId$.next(appId);
+        }
       }
     };
 

```